### ISSUE TITLE
Usernames containing "." causes launching tasks on GCP to fail

### ISSUE THREAD

**Issue by Key-Puncher** [skypilot-org/skypilot#2270]
https://github.com/skypilot-org/skypilot/blob/9b07ffc27cf8462c26aa359c4bf053e7d25aedcf/sky/templates/gcp-ray.yml.j2#L39

```I 07-19 14:54:23 cloud_vm_ray_backend.py:1700] Launching on GCP australia-southeast1 (australia-southeast1-a)
I 07-19 14:54:35 cloud_vm_ray_backend.py:730]
Got googleapiclient.errors.HttpError:
<HttpError 400 when requesting https://compute.googleapis.com/compute/v1/projects/alana-skypilot/zones/australia-southeast1-a/instances?alt=json returned "Invalid value for field 'resource.labels': ''.
Label value 'alana.huang' violates format constraints.
The value can only contain lowercase letters, numeric characters, underscores and dashes.
The value can be at most 63 characters long. International characters are allowed.".

Details: "[{'message': "Invalid value for field 'resource.labels': ''.
Label value 'alana.huang' violates format constraints.
The value can only contain lowercase letters, numeric characters, underscores and dashes.
The value can be at most 63 characters long. International characters are allowed.", 'domain': 'global', 'reason': 'invalid'}]">
```

Trying to use sky launch to create a new cluster on gcp, but my username contains a "." which is an invalid character for the gcp labels. I do not want to change my username, so can skypilot provide an option to overwrite the user field used in the gcp labels?

**Comment by Michaelvll** [comment]
Thanks for reporting this issue @Key-Puncher! We should definitely fix this issue, cc'ing @concretevitamin. We probably want to remove those invalid characters in the user name for the tags.

**Comment by saihtaungkham** [comment]
@Key-Puncher You could overwrite the local system user name with the environment variable.
`export SKYPILOT_USER="your-user-name"`

Hope this help.

**Comment by concretevitamin** [comment]
Fixed in #2290 by @saihtaungkham.
